openmdao/components/tests/test_meta_model_unstructured_comp.py

```diff
@@ -929,5 +927,52 @@ def test_metamodel_deprecated(self):
         prob.run_model()
         assert_rel_error(self, prob['mm.y1'], 1.5, 1e-2)
 
+    def test_metamodel_use_fd_if_no_surrogate_linearize(self):
```

Comment:
How does the component 'fd' itself when there isn't any code in the metamodel to declare that all derivs are to be calculated with the fd method? I think this exposes a hole in your test, and it is possible that either the function delete isn't working, or it takes more than just that to clean up some existing derivatives stuff (probably the partials dictionary needs to be cleaned out), but more pressing is fixing the test so that it doesn't give a false pass.

For the second half of the test, can you rebuild the test from scratch with a new Problem instance? That'll clear up any ambiguity about what is going on here.

Reply:
The method delete is definitely working. We should probably talk through this.